sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_auth_async.py

```diff
@@ -133,3 +133,24 @@ async def test_client_azure_sas_credential_async(self,
             batch = await producer_client.create_batch(partition_id='0')
             batch.add(EventData(body='A single message'))
             await producer_client.send_batch(batch)
+
+    @pytest.mark.liveTest
+    @pytest.mark.asyncio
+    async def test_client_azure_named_key_credential_async(live_eventhub):
+
+        hostname = "{}.servicebus.windows.net".format(eventhub_namespace.name)
+        consumer_client = EventHubConsumerClient(fully_qualified_namespace=live_eventhub['hostname'],
+                                                eventhub_name=live_eventhub['event_hub'],
+                                                consumer_group='$default',
+                                                credential=credential,
```

Comment:
where does `AzureNamedKeyCredential` get assigned to `credential`? Is it automatically created in a preparer and available, even though it hasn't been passed in to the test?

Reply:
line 141